httpcache.go

```diff
@@ -51,7 +55,48 @@ func CachedResponse(c Cache, req *http.Request) (resp *http.Response, err error)
 	}
 
 	b := bytes.NewBuffer(cachedVal)
-	return http.ReadResponse(bufio.NewReader(b), req)
+	returnResponse, err := http.ReadResponse(bufio.NewReader(b), req)
+	if err != nil {
+		return nil, fmt.Errorf("error loading response from cache: %s\n", err.Error())
+	}
+
+	rangeRaw := req.Header.Get("range")
+	if rangeRaw != "" {
+		tmp := strings.Split(rangeRaw, rangeTypeSeparator)
+		// standard format is bytes=START-END
+		rangetype, rangevalue := tmp[0], tmp[1]
+		if rangetype != "bytes" {
+			fmt.Printf("range type %s not supported\n", rangetype)
+			return returnResponse, nil
+		}
+		// we need to read all the body now, close it, and replace it with another reader
+		// as there is currently no way of "resetting" a Body
+		body, err := ioutil.ReadAll(returnResponse.Body)
+		if err != nil {
+			fmt.Printf("error reading cached response body: %s\n", err.Error())
+			return returnResponse, nil
+		}
+		returnResponse.Body.Close()
+		var rangedRequestStart, rangedRequestEnd int64
+		//TODO(uovobw): handle corrupted/nonstandard request header
+		rangeList := strings.Split(rangevalue, rangeSeparator)
+		// the range is in the form -VAL , the wanted range is (end-val)->end
+		if strings.HasPrefix(rangevalue, rangeSeparator) {
+			rangedRequestEnd = int64(len(body))
+			end, _ := strconv.ParseInt(rangeList[1], 10, 64)
```

Comment:
for the same reason i left the out the checks at line 81: if the request is broken it should get as it is to the server so that it might return a 400 or a relevant error, it's not the job of a caching proxy to validate these errors. the problem would be on the client sending a broken request and the error needs to bubble up on the server so that it can get fixed. just my opinion of course, it can be changed. what to do in that case though? removing the range header and letting the request through sounds wrong, giving an error before the server does too..suggestions?